qiskit/optimization/algorithms/grover_optimizer.py

```diff
@@ -345,11 +346,16 @@ def __init__(self, x: Union[List[float], np.ndarray], fval: float, variables: Li
             operation_counts: The counts of each operation performed per iteration.
             n_input_qubits: The number of qubits used to represent the input.
             n_output_qubits: The number of qubits used to represent the output.
+            intermediate_fval: The intermediate value of the objective function of the solution,
+                that is expected to be identical with ``fval``.
+            threshold: The threshold of Grover algorithm.
         """
-        super().__init__(x, fval, variables, None)
+        raw_results = {'intermediate_fval': fval, 'threshold': threshold}
```

Comment:
I guess you prefer to have these on the raw_results rather than properties on the grover result itself that would be more accessible/visible in the code/IDEs than key/value pairs - more so since these are not documented anywhere at present. If this is what is preferred it would be at least good to state these are provided in the raw result

Reply:
Which do you prefer to add them part of raw_results or add new properties?

Reply:
I try to make them new properties. It would be good to add comments on them.